practices/what_is_rest_api

```diff
@@ -0,0 +1,79 @@
+tags:
+  - Rest API
+  - Restful API
+  - API team
+  - API development
+  - Web development
+  - API design
+  - Http
+keywords:
+  - restful api
+  - http methods
+  - api endpoints
+  - json
+  - api design
+  - api documentation
+author_github_username: D-stevpm
+---
+
+# What is RESTful API
+
+![RESTful API](/files/what_is_rest_api.png)
+
+The majority of applications in use today are created for web and mobile platforms. Application Programming Interfaces (APIs) for the web have consequently gained importance. Among the different types of web APIs, REST (Representational State Transfer) API has become the most popular due to its simplicity, scalability, and flexibility.
+
+##What is an API 
+
+There have been a lot of definitions of API from industry leaders, after going through these numerous definitions I can simply describe APIs as a state of mechanisms that enable two software components to communicate with each other using a set of definitions and protocols. 
+
+According to ***Mastering API Architecture book by James Gough, Daniel Bryant & Matthew Auburn*** describe API as: 
+
+*  An API represents an abstraction of the underlying implementation.
+*  An API is represented by a specification that introduces types. Developers can understand the specifications and use tooling to generate code in multiple languages to implement an API consumer (software that consumes an API).
+*  An API has defined semantics or behavior to model the exchange of information effectively.
+*  Effective API design enables the extension to customers or third parties for business integration.
+
+##What is REST API?
+
+A set of constraints to be applied while developing web services are defined by the REST architectural style for software. In general, REST technology is favored above other, comparable technologies. This is frequently the case as REST requires less bandwidth and is therefore more suited for effective internet use.
+
+A RESTful API is based on representational state transfer (REST), an architectural style and approach to communications that is frequently used in the building of web services. It is also known as a RESTful web service or REST API. RESTful APIs are also referred to as ***web services***.
+
+Safe online information exchange between two computer systems is made possible by the RESTful API. To carry out various duties, the majority of business apps must interface with other internal and external applications. For instance, in order to automate invoicing and connect with an internal timesheet application, your internal accounts system must exchange information with your customers' banking system in order to generate monthly payslips. This information transmission is supported by RESTful APIs because they adhere to standards for safe, dependable, and effective software communication
+
+###How to Implement RESTful API
```

Comment:
Please add a section "Why You Might Want the REST API" and "Problems the REST API Helps to Solve" before this section. 

According to https://github.com/DXHeroes/knowledge-base-content/blob/master/CONTRIBUTING.md#content-composition